code/modules/vehicles/droppod.dm

```diff
@@ -0,0 +1,228 @@
+
+/obj/vehicle/droppod/Initialize()
+	. = ..()
+
+/obj/vehicle/droppod/emag_act()
+	return
+
+/obj/vehicle/droppod/Move()
+	return
+
+/obj/vehicle/droppod/verb/launchinterface()
+	set src in oview(1)
+	set category = "Vehicle"
+	set name = "Open launch interface"
+
+	ui_interact(usr)
+
+/obj/vehicle/droppod/load(var/mob/C) // this won't call the parent proc due to the differences and the fact it doesn't use load. Also only mobs can be loaded.
+	if(!ismob(C))
+		return 0
+	if(!isturf(C.loc)) 
+		return 0
+	if((humanload && passenger) || C.anchored)
+		return 0
+	if(humanload)
+		passenger = C
+	else
+		humanload = C
+
+	C.forceMove(src)
+
+	C.set_dir(dir)
+	C.anchored = 1
+
+	return 1
+
+/obj/vehicle/droppod/unload(var/mob/user, var/direction) // this also won't call the parent proc because it relies on load and doesn't expect a 2nd person
+	if(!(humanload || passenger))
+		return 0
+
+	var/turf/dest = null
+
+	if(direction)
+		dest = get_step(src, direction)
+	else if(user)
+		dest = get_turf(user)
+	if(!dest)
+		dest = get_step_to(src, get_step(src, turn(dir, 90)))
+
+	if(!dest || dest == get_turf(src))
+		var/list/options = new()
+		for(var/test_dir in alldirs)
+			var/new_dir = get_step_to(src, get_step(src, test_dir))
+			if(new_dir && user.Adjacent(new_dir))
+				options += new_dir
+		if(options.len)
+			dest = pick(options)
+		else
+			dest = get_turf(src)
+
+	if(!isturf(dest))
+		return 0
+
+	user.forceMove(dest)
+	user.set_dir(get_dir(loc, dest))
+	user.anchored = 0
+	user.pixel_x = initial(user.pixel_x)
+	user.pixel_y = initial(user.pixel_y)
+	user.layer = initial(user.layer)
+
+	if(user == humanload)
+		humanload = null
+	else
+		passenger = null
+	return 1
+
+/obj/vehicle/droppod/attack_hand(mob/user as mob)
+	..()
+	if(user == (humanload || passenger))
+		unload(user)
+	else
+		load(user)
+
+/obj/vehicle/droppod/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = default_state)
+	var/data[0]
+	data["used"] = used
+
+	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "droppod.tmpl", "Drop Pod", 400, 400, state = state)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/vehicle/droppod/Topic(href, href_list)
+	if(..())
+		return 1
+
+	if(href_list["fire"])
+		var/target = href_list["fire"]
+		if(target == "arrivals")
+			if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/fore)
+			else if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/port)
+			else if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/starboard)
+			else if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/aft)
+			else
+				firefromarea(/area/maintenance/arrivals)
+		else if(target == "cargo")
+			if(prob(65))
+				firefromarea(/area/quartermaster/loading)
+			else
+				firefromarea(/area/quartermaster/qm)
+		else if(target == "tcoms")
```

Comment:
I'm quite okay with them being able to get into TCOMs this easy, and if they do bomb it then administration can deal with them. But I'd like to see what others think as well.